x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.ts

```diff
@@ -0,0 +1,312 @@
+import moment from 'moment';
+
+import { RequestHandler, Logger } from '@kbn/core/server';
+import { TypeOf } from '@kbn/config-schema';
+import { CasesByAlertId } from '@kbn/cases-plugin/common/api/cases/case';
+import { AGENT_ACTIONS_INDEX } from '@kbn/fleet-plugin/common';
+import { CommentType } from '@kbn/cases-plugin/common';
+
+import {
+  HostIsolationRequestSchema,
+  responseActionBodySchemas,
+} from '../../../../common/endpoint/schema/actions';
+import { APP_ID } from '../../../../common/constants';
+import {
+  ISOLATE_HOST_ROUTE_V2,
+  RELEASE_HOST_ROUTE,
+  ENDPOINT_ACTIONS_DS,
+  ENDPOINT_ACTION_RESPONSES_DS,
+  failedFleetActionErrorCode,
+} from '../../../../common/endpoint/constants';
+import type {
+  EndpointAction,
+  EndpointActionData,
+  EndpointActionDataParameterTypes,
+  HostMetadata,
+  LogsEndpointAction,
+  LogsEndpointActionResponse,
+  ResponseActions,
+} from '../../../../common/endpoint/types';
+import type {
+  SecuritySolutionPluginRouter,
+  SecuritySolutionRequestHandlerContext,
+} from '../../../types';
+import { EndpointAppContext } from '../../types';
+import { getMetadataForEndpoints, FeatureKeys, getActionDetailsById } from '../../services';
+import { doLogsEndpointActionDsExists } from '../../utils';
+import { withEndpointAuthz } from '../with_endpoint_authz';
+
+export function registerResponseActionRoutes(
+  router: SecuritySolutionPluginRouter,
+  endpointContext: EndpointAppContext
+) {
+  const logger = endpointContext.logFactory.get('hostIsolation');
+
+  router.post(
+    {
+      path: ISOLATE_HOST_ROUTE_V2,
+      validate: HostIsolationRequestSchema,
+      options: { authRequired: true, tags: ['access:securitySolution'] },
+    },
+    withEndpointAuthz(
+      { all: ['canIsolateHost'] },
+      logger,
+      responseActionRequestHandler(endpointContext, 'isolate')
+    )
+  );
+
+  router.post(
+    {
+      path: RELEASE_HOST_ROUTE,
+      validate: HostIsolationRequestSchema,
+      options: { authRequired: true, tags: ['access:securitySolution'] },
+    },
+    withEndpointAuthz(
+      { all: ['canUnIsolateHost'] },
+      logger,
+      responseActionRequestHandler(endpointContext, 'unisolate')
+    )
+  );
+}
+
+const commandToFeatureKeyMap = new Map<ResponseActions, FeatureKeys>([
+  ['isolate', 'HOST_ISOLATION'],
+  ['unisolate', 'HOST_ISOLATION'],
+]);
+
+const endpointActionParametersTypesMap: Map<ResponseActions, EndpointActionDataParameterTypes> =
+  new Map([
+    ['isolate', undefined],
+    ['unisolate', undefined],
+  ]);
+
+function responseActionRequestHandler(
+  endpointContext: EndpointAppContext,
+  command: ResponseActions
+): RequestHandler<
+  unknown,
+  unknown,
+  TypeOf<typeof responseActionBodySchemas>,
+  SecuritySolutionRequestHandlerContext
+> {
+  return async (context, req, res) => {
+    const featureKey = commandToFeatureKeyMap.get(command) as FeatureKeys;
+    endpointContext.service.getFeatureUsageService().notifyUsage(featureKey);
```

Comment:
updated to handle different commands

Reply:
Should we check if there is a feature key before notify usage? I mean, is it possible to have a command that is not in the `commandToFeatureKeyMap` map? In that case, should we do something special?